Fix author display in items by making them into an actual array.

p-galligan opened the pull request:
So, while fiddling with index stuff, I noticed that the resources with multiple authors weren't displaying properly.

I tracked it down to the fact that `capture` only captures as a string, and that string didn't have any separators in it...

So, I made `author_list.html` return a string out of the authors array joined by '<', then captured the include, then split that captured variable to make it _back_ into an array.

This is all complicated because you for some reason can't use `assign` and an include in liquid. For sure let me know if you know of a better way.

Commit:
fix author display in items

Fix author display in items by making `author_list` a string separated by `<`, and then splitting the string to make an array.

helrond commented:
Hmm, this is somewhat convoluted. I'm wondering if we can do something similar to what's happening in `search_data.json` and just get rid of the `author_list.html` include. So, something like this in the `item.html` layout:

```
<ul class="list--unstyled"
{%- for agent in object.linked_agents -%}
  <li>{%- include linked_agent_title.html agent=agent -%}</li>
{%- endfor -%}
</ul>
```

Commit:
get object.linked_agents

Gets `object.linked_agents` instead of author list for iteration in item pages.

p-galligan commented:
Yep, that works. I would maybe suggest keeping the join in the `author_list.html` include though. I have changed it to `', '`. It's just a minor legibility change for the Author list in RACcess. Otherwise it just lists them all out with no space in between each one.

helrond commented:
OK, yup. Missed that the `author_list.html` include was being used in those inputs. In that case I'd suggest replacing the author list creation in `search_data.json` with a call to that include, since it's the same thing.

Commit:
Include author_list in search_data

Includes author_list for agents in `search_data.json` instead of using duplicate code.

p-galligan commented:
Removed the duplicate code in `search_data.json`
